src/win/util.c

```diff
@@ -107,29 +107,27 @@ int uv_utf8_to_utf16(const char* utf8Buffer, WCHAR* utf16Buffer,
 
 
 int uv_exepath(char* buffer, size_t* size_ptr) {
-  int utf8_len, utf16_buffer_len, utf16_len;
-  WCHAR* utf16_buffer;
+  WCHAR* utf16_buffer = 0;
+  DWORD utf16_buffer_len;
+  DWORD utf16_len;
+  int utf8_len;
+  HANDLE exe_handle = INVALID_HANDLE_VALUE;
   int err;
 
   if (buffer == NULL || size_ptr == NULL || *size_ptr == 0) {
     return UV_EINVAL;
   }
 
-  if (*size_ptr > 32768) {
-    /* Windows paths can never be longer than this. */
-    utf16_buffer_len = 32768;
-  } else {
-    utf16_buffer_len = (int) *size_ptr;
-  }
-
-  utf16_buffer = (WCHAR*) uv__malloc(sizeof(WCHAR) * utf16_buffer_len);
+  /* Windows paths can never be longer than 32768 characters. */
+  utf16_buffer_len = (DWORD)(*size_ptr < 32768 ? *size_ptr : 32768);
+  utf16_buffer = uv__malloc(sizeof(WCHAR) * utf16_buffer_len);
   if (!utf16_buffer) {
     return UV_ENOMEM;
   }
 
   /* Get the path as UTF-16. */
   utf16_len = GetModuleFileNameW(NULL, utf16_buffer, utf16_buffer_len);
-  if (utf16_len <= 0) {
+  if (utf16_len == 0) {
```

Comment:
Well, the function returns a DWORD, which is unsigned, so I'm not sure what the less than zero part was meant to signify? As I understand it the function returns zero on error and puts an explicit error code to be retrieved by GetLastError().